PowerPointLabs/PowerPointLabs/Utils/RegistryWatcher.cs

```
@@ -0,0 +1,99 @@
+        /// </summary>
+        public void Fire()
+        {
+            Notify();
+        }
+
+        public void Start()
+        {
+            watcher.Start();
+        }
+        public void Stop()
+        {
+            watcher.Stop();
+        }
+
+        public void SetValue(object o)
+        {
+            WindowsIdentity currentUser = WindowsIdentity.GetCurrent();
+            Registry.SetValue(String.Format("{0}\\{1}", currentUser.User.Value, path), key, o);
+        }
+
+        private void RegisterKeyChanged()
+        {
+            WindowsIdentity currentUser = WindowsIdentity.GetCurrent();
+            WqlEventQuery query = new WqlEventQuery(
+                     "SELECT * FROM RegistryValueChangeEvent WHERE " +
+                     "Hive = 'HKEY_USERS'" +
+                String.Format(@"AND KeyPath = '{0}\\{1}' AND ValueName='{2}'", currentUser.User.Value, path, key));
+            watcher = new ManagementEventWatcher(query);
+            watcher.EventArrived += (object sender, EventArrivedEventArgs e) => { Notify(); };
+        }
+
+        private T GetKeyAndUpdateKeyStatus()
+        {
+            using (RegistryKey key = Registry.CurrentUser.OpenSubKey(path))
+            {
+                object objectValue;
+                if (key == null || (objectValue = key.GetValue(this.key)) == null)
+                {
+                    throw new Exceptions.AssumptionFailedException("Key is null");
+                }
+                T result = (T)objectValue;
+                IsDefaultKey &= defaultKey == null || defaultKey.Equals(result);
+                return result;
+            }
+        }
+
+        private void Notify()
+        {
+            try
+            {
+                T key = GetKeyAndUpdateKeyStatus();
+                if (IsDefaultKey)
+                {
+                    return;
+                }
+                ValueChanged?.Invoke(this, key);
+            }
+            catch (Exception)
+            {
+
```

Empty catch clause. Should anything be done on an exception?